src/adapters/search_history.js

```diff
@@ -0,0 +1,40 @@
+import { get, set, del } from './store';
+
+const SEARCH_HISTORY_KEY = 'search_history';
+const HISTORY_SIZE = 100;
+
+// Add a query to the list
+export function save_query(q) {
+  // Delete query if it's already in the list
+  delete_query(q);
+
+  // Retrieve the search history
+  const searchHistory = get(SEARCH_HISTORY_KEY) || [];
+
+  // Put the query at the end of the array
+  searchHistory.push(q);
+
+  // Limit the list to the 100 last items
+  if (searchHistory.length > HISTORY_SIZE) {
+    searchHistory.shift();
+  }
+
+  // Serialize the list and save it in localStorage
+  set(SEARCH_HISTORY_KEY, searchHistory);
+}
+
+// Delete a query from the list
+export function delete_query(q) {
+  const searchHistory = get(SEARCH_HISTORY_KEY) || [];
+  const index = searchHistory.indexOf(q);
+  if (index > -1) {
```

Comment:
It would be better to early return on `-1` here, so we don't overwrite the history if it didn't change.